cpu/common/elf.h

```diff
@@ -6,8 +6,8 @@
 #endif
 
 #ifdef WORDS_BIGENDIAN
-#define ELF_SHORT_H
-#define ELF_LONG_H
+#define ELF_SHORT_H(ps) ((unsigned short)(ps))
+#define ELF_LONG_H(ps)  ((unsigned long)(ps))
```

Comment:
This changes the macro prototype, but I can't seen any change to the use of the macro below. Or is this just fixing a bug that was always there?

Reply:
It is fixing a bug that was always there. The old definition made the macro a no-op, but the cast is actually necessary, or else GCC warns for printf usage:
```
../../../cpu/common/parse.c:715:20: error: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'Elf32_Addr' {aka 'unsigned int'} [-Werror=format=]
  715 |           PRINTFQ (" vaddr: 0x%.8lx,", ELF_LONG_H (elf_spnt->sh_addr));
      |                    ^~~~~~~~~~~~~~~~~~             ~~~~~~~~~~~~~~~~~~~
      |                                                            |
      |                                                            Elf32_Addr {aka unsigned int}
```
The new prototype now matches the little-endian version of the macro.

Reply:
Good work